source/extensions/filters/http/basic_auth/basic_auth_filter.cc

```diff
@@ -31,18 +32,16 @@ FilterConfig::FilterConfig(UserMap&& users, const std::string& forward_username_
     : users_(std::move(users)), forward_username_header_(forward_username_header),
       stats_(generateStats(stats_prefix + "basic_auth.", scope)) {}
 
-bool FilterConfig::validateUser(absl::string_view username, absl::string_view password) const {
-  auto user = users_.find(username);
-  if (user == users_.end()) {
-    return false;
-  }
-
-  return computeSHA1(password) == user->second.hash;
-}
-
 BasicAuthFilter::BasicAuthFilter(FilterConfigConstSharedPtr config) : config_(std::move(config)) {}
 
 Http::FilterHeadersStatus BasicAuthFilter::decodeHeaders(Http::RequestHeaderMap& headers, bool) {
+  const auto* route_specific_settings =
+      Http::Utility::resolveMostSpecificPerFilterConfig<FilterConfigPerRoute>(decoder_callbacks_);
+  auto users = config_->users();
+  if (route_specific_settings != nullptr) {
+    users = route_specific_settings->users();
+  }
```

Comment:
I think a reference wrapper or something should be used here to avoid to copy the whole user map. It's too expensive for a single request.

Reply:
Thanks! I missed that due to `auto` and I didn't notice that it's not `UserMap&` as the method return type. I changed `users` to be a pointer.